providers/azure.go

```diff
@@ -220,6 +223,35 @@ func (p *AzureProvider) EnrichSession(ctx context.Context, session *sessions.Ses
 	return nil
 }
 
+func (p *AzureProvider) setUser(ctx context.Context, s *sessions.SessionState) {
```

Comment:
needs to return the errors
```suggestion
func (p *AzureProvider) setUser(ctx context.Context, s *sessions.SessionState) error {
```

Reply:
Furthermore, I would prefer if you move the method below the generic oidc methods. Probably best location would be between `redeemRefreshToken` and `makeAzureHeaders`.
This is just a preference for maintainability's sake and keeping a standard structure between the different provider files.